elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -473,6 +458,111 @@ defmodule API.Client.Channel do
   end
 
+  #############################################################
+  ##### Forwarding replies from the gateway to the client #####
+  #############################################################
+
+  # This the list of ICE candidates gathered by the gateway and relayed to the client
+  def handle_info(
+        {:ice_candidates, gateway_id, candidates, {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.ice_candidates",
+      attributes: %{
+        gateway_id: gateway_id,
+        candidates_length: length(candidates)
+      } do
+      push(socket, "ice_candidates", %{
+        gateway_id: gateway_id,
+        candidates: candidates
+      })
+
+      {:noreply, socket}
+    end
+  end
+
+  def handle_info(
+        {:invalidate_ice_candidates, gateway_id, candidates,
+         {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.invalidate_ice_candidates",
+      attributes: %{
+        gateway_id: gateway_id,
+        candidates_length: length(candidates)
+      } do
+      push(socket, "invalidate_ice_candidates", %{
+        gateway_id: gateway_id,
+        candidates: candidates
+      })
+
+      {:noreply, socket}
+    end
+  end
+
+  # DEPRECATED IN 1.4
+  # This message is sent by the gateway when it is ready to accept the connection from the client
+  def handle_info(
+        {:connect, socket_ref, resource_id, gateway_public_key, payload,
+         {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.connect", attributes: %{resource_id: resource_id} do
+      reply(
+        socket_ref,
+        {:ok,
+         %{
+           resource_id: resource_id,
+           persistent_keepalive: 25,
+           gateway_public_key: gateway_public_key,
+           gateway_payload: payload
+         }}
+      )
+
+      {:noreply, socket}
+    end
+  end
+
+  def handle_info(
+        {:connect, _socket_ref, resource_id, gateway_group_id, gateway_id, gateway_public_key,
+         preshared_key, ice_credentials, {opentelemetry_ctx, opentelemetry_span_ctx}},
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.connect", attributes: %{resource_id: resource_id} do
+      reply_payload = %{
+        resource_id: resource_id,
+        preshared_key: preshared_key,
+        client_ice_credentials: ice_credentials.client,
+        gateway_group_id: gateway_group_id,
```

Comment:
```suggestion
        site_id: gateway_group_id,
```

Gateway groups are sites in the product, right? Do we want to change this in the protocol here so both codebases can gradually move to the new domain term?

Reply:
@thomaseizinger site is meta-entity which is a combination of gateway group and resources connected to it. I think we should not do the rename right now and just keep our domain language the way it is now